# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546642872 28800
#      Fri Jan 04 15:01:12 2019 -0800
# Branch stable
# Node ID 58aa6d0694c21a5cbddb9dd0ab4c21332ad292dc
# Parent  183c6e9443b5104208944592ab11bc50c1d9ed8f
syminfo.cc: Fix static analyzer detected issues (bug #55347).

* syminfo.cc (symbol_info::display_line): Delete unreachable code.
* syminfo.cc (symbol_info_list::print_descriptor): Delete useless secondary
check on "param.modifer == 'c'" and re-indent code.
* syminfo.cc (symbol_info_list::parse_whos_line_format): Delete useless
variable error_encountered.

diff --git a/libinterp/corefcn/syminfo.cc b/libinterp/corefcn/syminfo.cc
--- a/libinterp/corefcn/syminfo.cc
+++ b/libinterp/corefcn/syminfo.cc
@@ -94,19 +94,16 @@ namespace octave
                     os << std::setiosflags (std::ios::left)
                        << std::setw (param.parameter_length);
                   }
                 break;
 
               default:
                 error ("whos_line_format: modifier '%c' unknown",
                        param.modifier);
-
-                os << std::setiosflags (std::ios::right)
-                   << std::setw (param.parameter_length);
               }
 
             switch (param.command)
               {
               case 'a':
                 {
                   char tmp[6];
 
@@ -275,35 +272,32 @@ namespace octave
                 os << std::setiosflags (std::ios::left)
                    << std::setw (param.parameter_length);
                 param_buf << std::setiosflags (std::ios::left)
                           << std::setw (param.parameter_length);
               }
 
             if (param.command == 's' && param.modifier == 'c')
               {
-                if (param.modifier == 'c')
-                  {
-                    int a = param.first_parameter_length - param.balance;
-                    a = (a < 0 ? 0 : a);
-                    int b = param.parameter_length - a - param.text.length ();
-                    b = (b < 0 ? 0 : b);
-                    os << std::setiosflags (std::ios::left) << std::setw (a)
-                       << "" << std::resetiosflags (std::ios::left) << param.text
-                       << std::setiosflags (std::ios::left)
-                       << std::setw (b) << ""
-                       << std::resetiosflags (std::ios::left);
-                    param_buf << std::setiosflags (std::ios::left)
-                              << std::setw (a)
-                              << "" << std::resetiosflags (std::ios::left)
-                              << param.line
-                              << std::setiosflags (std::ios::left)
-                              << std::setw (b) << ""
-                              << std::resetiosflags (std::ios::left);
-                  }
+                int a = param.first_parameter_length - param.balance;
+                a = (a < 0 ? 0 : a);
+                int b = param.parameter_length - a - param.text.length ();
+                b = (b < 0 ? 0 : b);
+                os << std::setiosflags (std::ios::left) << std::setw (a)
+                   << "" << std::resetiosflags (std::ios::left) << param.text
+                   << std::setiosflags (std::ios::left)
+                   << std::setw (b) << ""
+                   << std::resetiosflags (std::ios::left);
+                param_buf << std::setiosflags (std::ios::left)
+                          << std::setw (a)
+                          << "" << std::resetiosflags (std::ios::left)
+                          << param.line
+                          << std::setiosflags (std::ios::left)
+                          << std::setw (b) << ""
+                          << std::resetiosflags (std::ios::left);
               }
             else
               {
                 os << param.text;
                 param_buf << param.line;
               }
             os << std::resetiosflags (std::ios::left)
                << std::resetiosflags (std::ios::right);
@@ -426,17 +420,16 @@ namespace octave
     idx = 0;
     while (static_cast<size_t> (idx) < format_len)
       {
         whos_parameter param;
         param.command = '\0';
 
         if (format[idx] == '%')
           {
-            bool error_encountered = false;
             param.modifier = 'r';
             param.parameter_length = 0;
 
             int a = 0;
             int b = -1;
             int balance = 1;
             unsigned int items;
             size_t pos;
@@ -541,19 +534,17 @@ namespace octave
                                             : param.first_parameter_length);
             param.parameter_length = (a < 0
                                       ? 0
                                       : (param.parameter_length
                                          < param_length(pos_s)
                                          ? param_length(pos_s)
                                          : param.parameter_length));
 
-            // Parameter will not be pushed into parameter list if ...
-            if (! error_encountered)
-              params.push_back (param);
+            params.push_back (param);
           }
         else
           {
             // Text string, to be printed as it is ...
             std::string text;
             size_t pos;
             text = format.substr (idx, format.length ());
             pos = text.find ('%');
